### PR TITLE
Update openai to ^4.24.2 and include logprobs in `generationInfo`

### PR DESCRIPTION
**Pull request by functorism**
Update `openai` to `^4.24.2` and include logprobs in `generationInfo` in `ChatOpenAI` `_generate`

### CODE REVIEW

File: libs/langchain-openai/src/utils/azure.ts [hunk 1]
```diff
@@ -3,7 +3,7 @@ export interface OpenAIEndpointConfig {
   azureOpenAIApiInstanceName?: string;
   azureOpenAIApiKey?: string;
   azureOpenAIBasePath?: string;
-  baseURL?: string;
+  baseURL?: string | null;
```
Comment (jacoblee93) [inline libs/langchain-openai/src/utils/azure.ts:6]: This doesn't look necessary?

File: libs/langchain-openai/src/chat_models.ts [hunk 2]
```diff
@@ -596,9 +596,10 @@ export class ChatOpenAI<
             part.message ?? { role: "assistant" }
           ),
         };
-        if (part.finish_reason) {
-          generation.generationInfo = { finish_reason: part.finish_reason };
-        }
+        generation.generationInfo = {
+          ...(part.finish_reason ? { finishReason: part.finish_reason } : {}),
```
Comment (jacoblee93) [inline libs/langchain-openai/src/chat_models.ts:600]: Should be `finish_reason` in the `generationInfo` as well

File: libs/langchain-openai/src/utils/azure.ts (hunk 1 above)
Comment (functorism) [reply libs/langchain-openai/src/utils/azure.ts:6]: It's because of this change: https://github.com/openai/openai-node/commit/eb82824f89c31d752cdb4f198716a199e3b6eafd#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R26

But maybe it should be dealt with differently.

File: langchain-core/src/language_models/base.ts [hunk 3]
```diff
@@ -216,7 +216,7 @@ export interface FunctionDefinition {
    * To describe a function that accepts no parameters, provide the value
    * `{"type": "object", "properties": {}}`.
    */
-  parameters: Record<string, unknown>;
+  parameters?: Record<string, unknown>;
```
Comment (jacoblee93) [inline langchain-core/src/language_models/base.ts:219]: Core changes should be done in a separate PR

**Review by jacoblee93 — COMMENTED** [review]
Thanks so much for this!

File: langchain-core/src/language_models/base.ts (hunk 3 above)
Comment (functorism) [inline langchain-core/src/language_models/base.ts:219]: Likewise this change is because of: https://github.com/openai/openai-node/commit/c4fadb4513fc2e35cf0e4b9512ec39a6bd0e42dd#diff-30f3fcee562aad291d98f76d670f7cef5920b2ad946be42c3f6d02297767ec2cR772-R774

File: libs/langchain-openai/src/chat_models.ts (hunk 2 above)
Comment (functorism) [reply libs/langchain-openai/src/chat_models.ts]: Fixed :+1:

File: langchain-core/src/language_models/base.ts (hunk 3 above)
Comment (functorism) [reply langchain-core/src/language_models/base.ts:219]: Should I open a new PR with that singular change that gets merged before this one?

**Comment by functorism** [comment]
@jacoblee93 The api-refs build fine in the dev container now, a bit unclear to me why the deployment fails. I can't see any details.

File: langchain-core/src/language_models/base.ts (hunk 3 above)
Comment (jacoblee93) [reply langchain-core/src/language_models/base.ts:219]: Yes, if you wouldn't mind that would be really helpful!

**Comment by jacoblee93** [comment]
> @jacoblee93 The api-refs build fine in the dev container now, a bit unclear to me why the deployment fails. I can't see any details.

CI failure appears to be a flake, only blocker would be splitting the core change out separately (we will need to release a new version of core ahead of this). I can handle later today if you are busy!

**Comment by functorism** [comment]
@jacoblee93 If it's easy for you to slip such a thing in then that would be awesome. Much appreciated.

**Comment by jacoblee93** [comment]
Let's not change the internal types for now - I updated the OpenAI package to use the function definition type from `@langchain/core`. This is great though, will merge and cut a new release of `@langchain/openai`.

**Comment by jacoblee93** [comment]
Thank you! It's live in the latest version of `@langchain/openai`
